M2/Macaulay2/packages/AbstractSimplicialComplexes.m2

```diff
@@ -190,12 +190,17 @@ randomAbstractSimplicialComplex(ZZ,ZZ) := AbstractSimplicialComplex => (n,r) ->
 --  Cohen-Lenstra heuristics for torsion in homology of random complexes
 --  (Kahle, M. and Lutz, F. H. and Newman, A. and Parsons, K.).
 
-randomAbstractSimplicialComplex(ZZ,ZZ,ZZ) := AbstractSimplicialComplex => (n,m,d) -> (
+randomAbstractSimplicialComplex(ZZ,ZZ,ZZ) := AbstractSimplicialComplex =>  o -> (n,m,d) -> (
      L := for i from 1 to n list i;
      dDimlSubsets := subsets(L,d+1);
-     rdmFaces := for i from 1 to m list (dDimlSubsets#(random(binomial(n,d+1))));
-     append(append(rdmFaces,{L}),subsets(L,d));
-     abstractSimplicialComplex(rdmFaces))
+     rdmFaces := unique(for i from 1 to m list (dDimlSubsets#(random(binomial(n,d+1)))));
```

Comment:
I think you could use `randomSubset(dDimlSubsets, m)` here to guarantee that you get m distinct faces.  Even better than that would be `randomSubset(n, m)` and use the results with an unranking algorithm to generate only the subsets you need without having to call `subsets` to generate a bunch of subsets you won't use.

Also, why `Seed`?  If the option is meant to spend a bit of extra effort to verify that the output satisfies some condition, then maybe `Verify` would be more appropriate.